lib/OpenQA/Git.pm

```diff
@@ -3,29 +3,25 @@
 
 package OpenQA::Git;
 
-use Mojo::Base -base;
+use Mojo::Base -base, -signatures;
 use Cwd 'abs_path';
 use OpenQA::Utils qw(run_cmd_with_log_return_error);
 
 has 'app';
 has 'dir';
 has 'user';
 
-sub enabled {
-    my ($self) = @_;
+sub enabled ($self, $args = undef) {
     die 'no app assigned' unless my $app = $self->app;
     return ($app->config->{global}->{scm} || '') eq 'git';
 }
 
-sub config {
-    my ($self) = @_;
+sub config ($self, $args = undef) {
     die 'no app assigned' unless my $app = $self->app;
     return $app->config->{'scm git'};
 }
 
-sub _prepare_git_command {
-    my ($self, $args) = @_;
-
+sub _prepare_git_command ($self, $args = undef) {
```

Comment:
I know you added `$args = undef` here because the tests failed. However, it looks still wrong because if `$args` can be `undef` we should not unconditionally access a key of it like it is done in the next line. Since we're actually intentionally passing `undef` here in some cases you've found a bug. See my other inline comment for an easy fix.